src/System.IO.FileSystem/src/System/IO/UnixFileSystem.Linux.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace System.IO
+{
+    internal sealed partial class UnixFileSystem
+    {
+        public override bool CaseSensitive { get { return true; } }
```

Comment:
There's an issue tracking it already. It's such a corner case, I'd rather we not spend time on it now. I can add the issue # as a comment, though.